docs/explanations/architecture-2.md

```diff
@@ -0,0 +1,239 @@
+  constantly checks the resources declared in the cluster,
+  flagging the ones that no longer adhere with the deployed Kubewarden policies.
+
+## The journey of a Kubewarden policy
+
+### Default policy server
+
+On a new cluster, the Kubewarden components defined are:
+
+- the Custom Resource Definitions (CRD)
+- the `kubewarden-controller` Deployment
+- a `PolicyServer` Custom Resource named `default`.
+
+When the `kubewarden-controller` notices the default `PolicyServer` resource,
+it creates a Deployment of the `policy-server` component.
+
+Kubewarden works as a Kubernetes Admission Webhook.
+Kubernetes specifies using TLS to secure all Webhook endpoints.
+The `kubewarden-controller` sets up this secure communication
+by:
+
+1. Generating a self-signed Certificate Authority
+1. Use this CA to generate a TLS certificate key for the `policy-server` Service.
+
+These objects are all stored as `Secret` resources in Kubernetes.
+
+Finally, `kubewarden-controller` creates the `policy-server` Deployment
+and a Kubernetes ClusterIP Service
+to expose it inside the cluster network.
+
+### Defining the first policy
+
+This diagram shows what happens when defining the first policy
+bound to the default `policy-server` in the cluster:
+
+:::note
+
+A policy must define which Policy Server it must run on.
+Or, we say it binds to a Policy Server.
+You can have different policies with the same Wasm module and settings running in many Policy Servers.
+However, you can't have a single policy definition that runs in many policy servers.
+
+:::
+
+The `kubewarden-controller` notices the new `ClusterAdmissionPolicy` resource and
+so finds the bound `PolicyServer` and reconciles it.
+
+### Reconciliation of `policy-server`
+
+When creating, modifying or deleting a `ClusterAdmissionPolicy`,
+a reconciliation loop activates in `kubewarden-controller`,
+for the `PolicyServer` owning the policy.
+This reconciliation loop creates a ConfigMap with all the polices bound to the `PolicyServer`.
+Then the Deployment rollout of the `policy-server` starts.
+It results in starting the new `policy-server` instance with the updated configuration.
+
+At start time, the `policy-server` reads its configuration from the ConfigMap
+and downloads all the Kubewarden policies specified.
+You can download Kubewarden policies from remote HTTP servers and container registries.
+
+You use policy settings parameters to tune a policies' behavior.
+After startup and policy download the `policy-server`
+checks the policy settings provided by the user are valid.
+
+The `policy-server` validates policy settings by invoking
+the `validate_setting` function exposed by each policy.
+There is further documentation in the
+[specification reference](../reference/spec/01-intro-spec.md)
+section of the documentation.
+
+The `policy-server` exits with an error
+if one or more policies received wrong configuration parameters
+from the policy specification provided by the user.
```

Comment:
This is not true anymore. In a recent change, the policy server has been changed to start even if with a invalid policy. What happens now is that all the admission request sent to be evaluated by the invalid policy will return an error.